pkg/apis/trust/v1alpha1/types_bundle.go

```diff
@@ -176,8 +170,15 @@ type BundleCondition struct {
 	// with respect to the current state of the Bundle.
 	// +optional
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// SourceVersions is a list of the versions of the sources which were used to
+	// generate the Bundle target. The index of the source version corresponds to
+	// the index of the source in the BundleSpec.
+	SourceVersions []string `json:"sourceVersions,omitempty"`
```

Comment:
comment: I think I'm negative on  this change overall and part of that is "The index of the source version corresponds to the index of the source in the BundleSpec." If we're using `strings`, going by index makes it impossible to check any version for a source on a Bundle without also parsing the spec. We could define some sort of format for the version strings, but with both approaches the UX isn't great.

Aside from that, 99% of the time, the only version I'd care about or want to check is the default CAs version. I'd want to check it regularly, and update my default CA package if it's out of date.

The versions of Secrets / CMs seem useful to have on the status, because they'll make it clear what the last successfully synchronized version of the Secret / CM was, at a glance - but if the Bundle has failed to update I'd expect there to be error messages and a "condition" indicating failure. I don't need to check the versions - I just need to know that it failed and why.

I don't have control over which version of a Secret my bundle syncs. I only have control over whether a given Secret is synced. As such, the version information is only relevant in the usually very short period of time between a Secret/CM being updated and being synced to a target.

The default CA package version is fundamentally different to Secret/CM/InLine. I think its version needs to be highlighted very clearly!